### PR TITLE
Fix vector interface on GPUs

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Summary

It turns out the vector interface on GPUs was only working on Power9s thanks to undefined behavior. This PR fixes it. The issue was that the `*this` pointer was being captured in the lambdas in the portable for in the base class.

To rectify, I dereference the `*this` pointer to make a copy of the derived class to pass into the `portableFor`.

To correctly test this, we need a CI build matrix for x86-volta nodes. I have added one, but some extra eyes from, e.g., @ktsai7, @junghans , or @dholladay00  would be welcome.

## PR Checklist

- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Format your changes by using the `make format` command after configuring with `cmake`.
- [x] Document any new features, update documentation for changes made.
- [x] Make sure the copyright notice on any files you modified is up to date.

### CODE REVIEW

**Review by junghans — APPROVED** [review]

File: singularity-eos/eos/CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,7 @@ endif()
 
 set(EOS_SRCS
     eos.hpp
+    eos_base.hpp
```
Comment (jhp-lanl) [inline singularity-eos/eos/CMakeLists.txt:21]: Thanks for adding this!

**Review by jhp-lanl — APPROVED** [review]
Changes look good once the x86 starts passing

**Comment by Yurlungur** [comment]
The test issue is related to spack buildouts on the CI. Fixing it correctly will take some time, which @ktsai7 offered to commit. In the meantime, I will jury-rig something.

**Comment by Yurlungur** [comment]
This finally passes. In a future MR, @ktsai7 will submit a fix to the CI. For now, merging with my jury-rigged solution.

**Pull request merged**